### PR TITLE
Feature/unpublish refactor

### PR DESCRIPTION
**Pull request by kafkaphoenix**
## Description

Refactor unpublish

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: engine/admin-api/domain/usecase/version/unpublish.go [hunk 1]
```diff
@@ -17,39 +16,46 @@ func (h *Handler) Unpublish(
...
-	if v.Status != entity.VersionStatusPublished {
-		return nil, internalerrors.ErrInvalidVersionStatusBeforeUnpublishing
+	if vers.Status != entity.VersionStatusPublished {
+		h.registerActionFailed(user.ID, productID, vers, ErrVersionCannotBeUnpublished, "unpublish")
+		return nil, ErrVersionCannotBeUnpublished
 	}
 
-	err = h.k8sService.Unpublish(ctx, productID, v)
+	err = h.k8sService.Unpublish(ctx, productID, vers)
 	if err != nil {
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/unpublish.go:42]: missing registerActionFailed

File: engine/admin-api/domain/usecase/version/unpublish.go [hunk 2]
```diff
@@ -17,39 +16,46 @@ func (h *Handler) Unpublish(
...
 
-	v.PublicationAuthor = nil
-	v.PublicationDate = nil
-	v.Status = entity.VersionStatusStarted
+	vers.PublicationAuthor = nil
+	vers.PublicationDate = nil
+	vers.Status = entity.VersionStatusStarted
 
-	err = h.versionRepo.Update(productID, v)
+	err = h.versionRepo.Update(productID, vers)
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/unpublish.go:50]: this should be non blocking

File: engine/admin-api/domain/usecase/version/unpublish.go [hunk 3]
```diff
@@ -17,39 +16,46 @@ func (h *Handler) Unpublish(
...
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error updating version %q: %w", vers.Tag, err)
 	}
 
-	err = h.userActivityInteractor.RegisterUnpublishAction(user.ID, productID, v, comment)
+	err = h.userActivityInteractor.RegisterUnpublishAction(user.ID, productID, vers, comment)
```
Comment (alexrodfe) [inline engine/admin-api/domain/usecase/version/unpublish.go:55]: this also should be non blocking

**Review by alexrodfe — CHANGES_REQUESTED** [review]
Fantastic, I have some comments to improve logic and follow other file's structure

File: engine/admin-api/domain/usecase/version/unpublish.go (hunk 3 above)
Comment (kafkaphoenix) [reply engine/admin-api/domain/usecase/version/unpublish.go:61]: done

File: engine/admin-api/domain/usecase/version/unpublish.go (hunk 1 above)
Comment (kafkaphoenix) [reply engine/admin-api/domain/usecase/version/unpublish.go:41]: done

**Review by alexrodfe — DISMISSED** [review]

**Review by alexrodfe — APPROVED** [review]

**Pull request merged**